homeassistant/components/leneda/const.py

```diff
@@ -0,0 +1,139 @@
+"""Constants for the Leneda integration."""
+
+from datetime import timedelta
+
+from leneda.obis_codes import ObisCode
+
+DOMAIN = "leneda"
+
+CONF_API_TOKEN = "api_token"
+CONF_ENERGY_ID = "energy_id"
+CONF_METERING_POINTS = "metering_points"
+
+SCAN_INTERVAL = timedelta(hours=1)
+
+# Sensor types and their corresponding OBIS codes
+SENSOR_TYPES = {
```

Comment:
The state class and device class can all come from Home Assistant's enums.

`from homeassistant.components.sensor import SensorStateClass, SensorDeviceClass`

Do we also want to specify a unit of measurement here?

These could also be a list of `SensorEntityDescription` objects in the future (https://developers.home-assistant.io/docs/core/entity?_highlight=sensorentitydescription#example)

Reply:
I have amended the code to use `SensorStateClass`, `SensorDeviceClass` from `homeassistant.components.sensor`.

The unit of measurement is currently extracted through via the leneda-client lib, but they could in fact be included here for convenience. If it is not strictly required, I would leave it as-is for the moment and improve upon this in a later iteration.

Reply:
The icons could also be moved into a separate icons.json file.